node/sync/src/block_sync.rs

```diff
@@ -215,6 +215,13 @@ impl<N: Network> BlockSync<N> {
         let block_requests = self.prepare_block_requests();
         trace!("Prepared {} block requests", block_requests.len());
 
+        // If there are no block requests, then try if we can advance with responses we already
+        // have in the sync pool.
+        if block_requests.is_empty() && !self.responses.read().is_empty() {
+            trace!("No block requests to send - try handling responses we already have");
+            self.handle_responses(self.canon.latest_block_height());
```

Comment:
Shouldn't we `return Ok(());` after handling the responses here?

Reply:
The method doesn't return anything, so not sure why you say that?

Reply:
We could return early, but it's not really necessary as we know that the `block_requests` is empty.

Reply:
> The method try_block_sync doesn't return anything

ah yeah, then just `return;`

Reply:
> it's not really necessary as we know that the block_requests is empty.

indeed, I only mean it as a readability improvement